### ISSUE TITLE
seed / destroy race condition prevent destruction

### ISSUE THREAD

**Issue by mvayngrib** [feross/webtorrent#254]
to repro:
    
    client.seed()
    client.destroy()

seed() calls self.add asynchronously and ends up adding to self.torrents after destroy empties it

I suggest something like bittorrent-dht has - self._destroyed with checks, and can implement it if you like. However, this will likely force changes to the **ontorrent** callback signature. Currently it's not a traditional node callback - ontorrent(torrent), but it will likely need to become one, a la:
  
    if (self._destroyed) ontorrent && ontorrent(new Error('client is destroyed'))

**Comment by feross** [comment]
Good catch. I'll fix this up.

We don't actually need to make the `ontorrent` take an error callback. Calling `destroy` just means that the 'torrent' event will never get emitted. That's how it works with sockets and the 'listen' event in node.js. For example:

```js
var net = require('net')
var socket = net.connect(4000, function () {
  console.log('listen')
})
socket.destroy()
```

will not log `listen`.

**Comment by feross** [comment]
Fixed this issue, added tests, and released as 0.27.0.
